src/librustc_borrowck/borrowck/mir/elaborate_drops.rs

```diff
@@ -672,27 +673,45 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                 self.drop_ladder(c, fields)
             }
             _ => {
-                let variant_drops : Vec<BasicBlock> =
-                    (0..adt.variants.len()).map(|i| {
-                        self.open_drop_for_variant(c, &mut drop_block,
-                                                   adt, substs, i)
-                    }).collect();
-
+                let mut values = Vec::with_capacity(adt.variants.len());
+                let mut blocks = Vec::with_capacity(adt.variants.len() + 1);
```

Comment:
Good observation. I think why this works now is because the last element in `values` array gets ignored and the last block still acts as an `otherwise` branch.